railties/test/application/rake/migrations_test.rb

```diff
@@ -206,6 +278,46 @@ class AMigration < ActiveRecord::Migration::Current
         end
       end
 
+      test "raise error on any move when target migration does not exist" do
+        app_file "db/migrate/01_one_migration.rb", <<-MIGRATION
```

Comment:
I'd like to see a test somewhere (assuming we don't already have one, though I haven't checked) to ensure that `VERSION=01_one_migration.rb` will work, even after we make `VERSION=1.2` fail.

Reply:
> I'd like to see a test somewhere (assuming we don't already have one, though I haven't checked) to ensure that VERSION=01_one_migration.rb will work, even after we make VERSION=1.2 fail.

I just implemented this behavior and added test cases.